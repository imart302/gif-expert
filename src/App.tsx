import { useRef, useState } from 'react';
import AddCategory from './components/AddCategory';
import GifGrid from './components/GifGrid';
import './styles.css'

function App() {
  const [categories, setCategories] = useState<{ categories: string[], categoriesL: string[]}>({categories: [], categoriesL:[]});
  const inputRef = useRef<HTMLInputElement>(null);

  const onAddCategory = (cat: string) => {
    if (cat) {
      if (!categories.categoriesL.includes(cat.toLowerCase())) {
        setCategories({categories: [...categories.categories, cat], categoriesL: [...categories.categoriesL, cat.toLowerCase()]});
       
      }
    }
  };

  return (
    <div className="App">
      <h1>Gif expert app</h1>
      <br></br>
      <AddCategory onNewCategory={onAddCategory}></AddCategory>

      {categories.categories.map((cate, id) => (
        <div key={id}>
          <br></br>
          <GifGrid category={cate}></GifGrid>
        </div>
      ))}
    </div>
  );
}

export default App;
