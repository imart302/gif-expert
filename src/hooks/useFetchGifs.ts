

import React, { useEffect, useState } from 'react'
import { GiphyData, GiphyResponse } from '../api/giphyInterfaces/search'
import { searchGiphy } from '../api/giphySearch';

export default function useFetchGifs(category: string) {
  
  const [gifs, setGifs] = useState<GiphyData[]>([]);
  const [isLoading, setIsLoading] = useState<boolean>(true);
  
  const getGifs =async () => {
    const resp = await searchGiphy(category);
    setGifs(resp.data);
    setIsLoading(false);
  }

  useEffect(() => {
    getGifs();
  });

  return (
    {
      gifs,
      isLoading,
    }
  )
}
