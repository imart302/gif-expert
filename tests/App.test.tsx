import { fireEvent, render, screen } from "@testing-library/react";
import React from "react";
import { vi } from "vitest";
import App from '../src/App';
import useFetchGifs from "../src/hooks/useFetchGifs";


vi.mock("../src/hooks/useFetchGifs");

describe('Tests on <App>', () => {

  afterEach(() => {
    vi.resetAllMocks();
  })

  test('should match snapshot', () => {

    const {container} = render(<App></App>);

    expect(container).toMatchSnapshot();

  });


  test('should add category on form submit', () => {

    vi.mocked(useFetchGifs).mockReturnValue({
      gifs: [],
      isLoading: false
    });
    const { container } = render(<App></App>);

    const form = screen.getByRole<HTMLFormElement>('form');
    const input = screen.getByRole<HTMLInputElement>('textbox');

    const category = 'Hello';

    fireEvent.input(input, {target: {value: category}});
    fireEvent.submit(form);

    const hello = screen.getByText<HTMLHeadElement>(category);
    expect(hello).toBeDefined();
    

  });

});