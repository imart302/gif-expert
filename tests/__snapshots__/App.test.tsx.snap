// Vitest Snapshot v1

exports[`Tests on <App> > should match snapshot 1`] = `
<div>
  <div
    class="App"
  >
    <h1>
      Gif expert app
    </h1>
    <br />
    <form
      aria-label="form"
    >
      <input
        placeholder=""
        type="text"
        value=""
      />
    </form>
  </div>
</div>
`;
